### ISSUE TITLE
Markup submodule support

### ISSUE THREAD

**Issue by SKY-ALIN** [telefone-org/framework#13]
Hello! I want to table a proposal to add a special submodule to generate Telegram-specific markup easily.

## Motivation

A common markdown format that every programmer uses is really simple and doesn’t require special utils for that, but MarkdownV2 and HTML which let include spoiler, strikethrough and other telegram innovations are different. For example, MarkdownV2 also requires to escape 18 characters, and if it’s not, reject the request. This may make the task of designing markup generation for bots that include business logic complicated enough and I suppose it’s better to have a ready-made out-of-the-box solution for this.

### General explanation

I have recently worked on a related challenge and I’m writing this to propose to re-use my solution. It’s the [`telegram-text`](https://github.com/SKY-ALIN/telegram-text) module. The general structure in the context of the `telefone` module I propose to implement is the following:

- Add [`telegram-text`](https://github.com/SKY-ALIN/telegram-text) as a dependency (maybe even optional)
- Add support for message objects rendering depending on format mode (Markdown / MarkdownV2 / HTML) inside the message sending function
- Add imports into the `telefone` module core to import styles and other needed classes as `from telefone import ...`. Or it’s quite pretty to add a separated file (`text.py` or `markup.py` for example) to import as `from telefone.text import ...`

### Why [`telegram-text`](https://github.com/SKY-ALIN/telegram-text) module?

- Include all rules for all markup modes
- Almost entirely covered by tests
- Fully documented
- Open to contribute
- Has a few presets and is easily expandable to add new

Example of the module:

```python
from telegram_text import Bold, Italic, Underline

text = Underline(Bold("Bold") + "and" + Italic("italic") + "with underline.")
```

<p align="center">
  <img 
    width="600"
    src="https://telegram-text.alinsky.tech/_images/basic_example_result.jpg"
  />
</p>

### Usage example in the `telefone` context

```python
from telefone import Bot
from telefone.text import Spoiler

bot = Bot("your-token")


@bot.on.message()
async def handler(_):
    return Spoiler("Hello world!")

bot.run_forever()
```

**Comment by exthrempty** [comment]
Hello! Thanks for the proposal. I think your approach is great and pretty well done, but I have a question.

Can I count on this project being maintained? Looks like it's almost complete and fully functional. We are adopting solutions that will lay the foundation for the first stable release, so they should be actively maintained.

The specifics of what we are making here prevent me from making this package optional. The work needs to be done in some tight places, where adding auxiliary checks seems unreasonable.

Sadly, I don't have the capacity to work at this at the moment, but I'll think of an implementation and get started on it as soon as possible. If you have any ideas, feel free to reply.

**Comment by SKY-ALIN** [comment]
Thank you for your reply. Yes, this project will be maintained, it's open to contribute and sure you can join at any time! Also, I can help to integrate `telegram-text` into `telefone` if you agree with the proposed interface. The integration doesn't look complicated to implement.

**Comment by exthrempty** [comment]
Yes, I agree, seems pretty natural to me. The thing is, it may be hard for you to help because the framework is entirely undocumented. I plan to document all interfaces in the upcoming version, though.
